From e9de368f0952f76a38a2e9d4dcfa0abb66717bdf Mon Sep 17 00:00:00 2001
From: "Burla, Satananda" <sburla@marvell.com>
Date: Mon, 5 Oct 2020 09:16:25 -0700
Subject: [PATCH 0776/1921] octeontx2-dpi: Fix DPI engine blks allocation

Octeontx2 support 64KB fifo, increase allocation

Signed-off-by: Burla, Satananda <sburla@marvell.com>
Change-Id: Idc675af2c182f2d6c5bef325ad450ff8a7193cf2
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37348
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit 68e1b99442db9924da94d971f09eb0c0daaae520)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37423
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx2-dpi/dpi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-dpi/dpi.c b/drivers/soc/marvell/octeontx2-dpi/dpi.c
index 2a3537ea2e8c..76a2c02c302a 100644
--- a/drivers/soc/marvell/octeontx2-dpi/dpi.c
+++ b/drivers/soc/marvell/octeontx2-dpi/dpi.c
@@ -140,9 +140,9 @@ static int dpi_init(struct dpipf *dpi)
 
 	for (engine = 0; engine < dpi_dma_engine_get_num(); engine++) {
 		if (engine == 4 || engine == 5)
-			reg = DPI_ENG_BUF_BLKS(8);
+			reg = DPI_ENG_BUF_BLKS(16);
 		else
-			reg = DPI_ENG_BUF_BLKS(4);
+			reg = DPI_ENG_BUF_BLKS(8);
 
 		dpi_reg_write(dpi, DPI_ENGX_BUF(engine), reg);
 
-- 
2.31.1

